src/Microsoft.OData.Core/UriParser/Parsers/ExpandOptionParser.cs

```diff
@@ -134,6 +150,13 @@ internal List<ExpandTermToken> BuildExpandTermToken(PathSegmentToken pathToken,
                     string text = this.enableCaseInsensitiveBuiltinIdentifier
                         ? this.lexer.CurrentToken.Text.ToLowerInvariant()
                         : this.lexer.CurrentToken.Text;
+
+                    // Prepend '$' prefix if needed.
+                    if (this.enableNoDollarQueryOptions && !text.StartsWith(UriQueryConstants.DollarSign))

```

Comment:
Code inspection will also report an error for this StartsWith (also for line 538 in DataQueryOptionParser)

Because the behavior of 'string.StartsWith(string)' could vary based on the current user's locale settings, replace this call in 'ExpandOptionParser.BuildExpandTermToken(PathSegmentToken, string)' with a call to 'string.StartsWith(string, StringComparison)'. If the result of 'string.StartsWith(string, StringComparison)' will be displayed to the user, such as when sorting a list of items for display in a list box, specify 'StringComparison.CurrentCulture' or 'StringComparison.CurrentCultureIgnoreCase' as the 'StringComparison' parameter. If comparing case-insensitive identifiers, such as file paths, environment variables, or registry keys and values, specify 'StringComparison.OrdinalIgnoreCase'. Otherwise, if comparing case-sensitive identifiers, specify 'StringComparison.Ordinal'. (#BuildExpandTermToken(Microsoft.OData.Core.UriParser.Syntactic.PathSegmentToken,System.String))

So, update this to !text.StartsWith(UriQueryConstants.DollarSign, StringComparison.Ordinal)